sdk/provisioning/Azure.Provisioning/src/Extensions.cs

```diff
@@ -23,7 +24,12 @@ public static string ToCamelCase(this string str)
         public static bool IsChildResource(this Resource resource)
         {
             //TODO: this is a bit of a hack. We should probably have a better way to determine if a resource is a child resource
-            return resource is DeploymentScript || (resource.Parent is not null && resource.Parent is not ResourceGroup && resource.Parent is not Subscription);
+            return resource.Parent is not null &&
+                   resource.Parent is not ResourceGroup &&
+                   resource.Parent is not Subscription &&
+                   resource is not DeploymentScript &&
+                   resource is not Subscription &&
+                   resource is not RoleAssignment;
```

Comment:
lets unravel this hack vs adding another use case for it.

Originally this was used to determine which file you went into.  This should be resolved with the module constructs that we translate everything into before writing.

Reply:
this was an unrelated change for the roleassignment work I was doing.